R/oauth-refresh.R

```diff
@@ -17,6 +17,9 @@ refresh_oauth2.0 <- function(endpoint, app, credentials) {
   )
 
   response <- POST(refresh_url, body = body, encode = "form")
+  if (has_oauth2.0_error(response)) {
+    return(NULL)
```

Comment:
My thinking was that here in `refresh_oauth2.0` our goal is to get new, valid credentials using our refresh token (obviously), and OAuth2 provides a defined mechanism to inform us of that this can't happen - by returning `invalid_grant` in the error response.

Separating this out from `stop_for_status` allows us to still catch any errors that occur - 404s, redirects, internal server errors, etc. All of these are errors that prevent us from getting refreshed credentials, but there isn't any specific information in them that tells us what to do.

Contrast that with the OAuth2 error(s) that tell us we have an invalid token, which is why our logic is to clear the token from our cache.

Taking it a step further, the OAuth2 spec is pretty clear that `invalid_grant` is the error we should look for in the case of the refresh token. I've currently written this in such a way that `has_oauth2.0_error` will return `TRUE` even in the case of OAuth2 errors that aren't specific to the refresh token.

I did opt to cast a broader net, but there's an argument to be made that this function should be called `has_oauth2.0_refresh_token_error(response)` and only return `TRUE` when `invalid_grant` is present.

Thoughts?